src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (a = null) => {
+    if (a === null) {
+      const newResult = result;
+
+      result = 0;
+
+      return newResult;
+    } else {
+      result += a;
```

Comment:
```suggestion
      return newResult;
    }
    
    result += a;
```
No need to add `else` here, as your code will not go further after `return` anyway